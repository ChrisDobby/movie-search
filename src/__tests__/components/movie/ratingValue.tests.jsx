import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import RatingValue from '../../../components/movie/ratingValue';

describe('ratingStar', () => {
    it('should call select when clicked', () => {
        const props = {
            select: jest.fn(),
            value: 1,
            currentRating: 0,
        };
        const component = shallow(<RatingValue {...props} />);

        component.find('div').at(0).simulate('click');

        expect(props.select.mock.calls).toHaveLength(1);
    });

    it('should call select when return key pressed', () => {
        const props = {
            select: jest.fn(),
            value: 1,
            currentRating: 0,
        };
        const component = shallow(<RatingValue {...props} />);

        component.find('div').at(0).simulate('keydown', { keyCode: 13 });

        expect(props.select.mock.calls).toHaveLength(1);
    });

    it('should not call select when other key pressed', () => {
        const props = {
            select: jest.fn(),
            value: 1,
            currentRating: 0,
        };
        const component = shallow(<RatingValue {...props} />);

        component.find('div').at(0).simulate('keydown', { keyCode: 65 });

        expect(props.select.mock.calls).toHaveLength(0);
    });

    it('should render correctly when not selected', () => {
        const props = {
            select: jest.fn(),
            value: 10,
            currentRating: 5,
        };

        const component = renderer.create(<RatingValue {...props} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should render correctly when selected', () => {
        const props = {
            select: jest.fn(),
            value: 4,
            currentRating: 5,
        };

        const component = renderer.create(<RatingValue {...props} />);

        expect(component.toJSON()).toMatchSnapshot();
    });
});
