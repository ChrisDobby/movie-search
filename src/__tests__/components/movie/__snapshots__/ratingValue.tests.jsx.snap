// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ratingStar should render correctly when not selected 1`] = `
<div
  className="col-xs-3 col-sm-2 col-md-2 col-lg-1"
  onClick={[MockFunction]}
  onKeyDown={[Function]}
  role="presentation"
  style={
    Object {
      "background": "#ffffff",
      "border": "2px solid #ffeeba",
      "borderRadius": "50%",
      "color": "#856404",
      "cursor": "pointer",
      "fontSize": "32px",
      "margin": "4px",
      "maxHeight": "80px",
      "maxWidth": "80px",
      "padding": "8px",
      "textAlign": "center",
    }
  }
>
  10
</div>
`;

exports[`ratingStar should render correctly when selected 1`] = `
<div
  className="col-xs-3 col-sm-2 col-md-2 col-lg-1"
  onClick={[MockFunction]}
  onKeyDown={[Function]}
  role="presentation"
  style={
    Object {
      "background": "#fff3cd",
      "border": "2px solid #ffeeba",
      "borderRadius": "50%",
      "color": "#856404",
      "cursor": "pointer",
      "fontSize": "32px",
      "margin": "4px",
      "maxHeight": "80px",
      "maxWidth": "80px",
      "padding": "8px",
      "textAlign": "center",
    }
  }
>
  4
</div>
`;
